projects/core/src/cart/cart.module.ts

```diff
@@ -1,18 +1,24 @@
-import { ModuleWithProviders, NgModule } from '@angular/core';
+import { APP_INITIALIZER, ModuleWithProviders, NgModule } from '@angular/core';
 import { PageMetaResolver } from '../cms/page/page-meta.resolver';
 import { ActiveCartService } from './facade/active-cart.service';
 import { CartDataService } from './facade/cart-data.service';
 import {
   CartService,
   CartVoucherService,
-  WishListService,
   SelectiveCartService,
+  WishListService,
 } from './facade/index';
 import { MultiCartService } from './facade/multi-cart.service';
 import { CartPageMetaResolver } from './services/cart-page-meta.resolver';
+import { MultiCartStatePersistenceService } from './services/multi-cart-state-persistence.service';
 import { CartStoreModule } from './store/cart-store.module';
 import { MultiCartStoreModule } from './store/multi-cart-store.module';
 
+function cartStatePersistenceFactory(cartStatePersistenceService): any {
+  const result = () => cartStatePersistenceService;
```

Comment:
What about calling `.sync()` explicitly after the context config is initialized (instead of implicit call in the constructor)? It could help us avoid race condition and working around it.
```typescript
 function cartStatePersistenceFactory(
   cartStatePersistenceService: MultiCartStatePersistenceService,
   configInit: ConfigInitializerService): any { 
      configInit.getStableConfig('context').then(() => {
         cartStatePersistenceService.sync();
       });
  }

/*...*/

  deps: [MultiCartStatePersistenceService, ConfigInitializerService], 
```